lib/monitoring/aws-rds/AuroraClusterMonitoring.ts

```diff
@@ -0,0 +1,239 @@
+import {
+  GraphWidget,
+  HorizontalAnnotation,
+  IWidget,
+} from "aws-cdk-lib/aws-cloudwatch";
+
+import {
+  RdsClusterMetricFactory,
+  RdsClusterMetricFactoryProps,
+} from "./RdsClusterMetricFactory";
+import {
+  BaseMonitoringProps,
+  ConnectionAlarmFactory,
+  CountAxisFromZero,
+  DefaultGraphWidgetHeight,
+  HalfWidth,
+  HighConnectionCountThreshold,
+  LowConnectionCountThreshold,
+  MetricWithAlarmSupport,
+  Monitoring,
+  MonitoringScope,
+  PercentageAxisFromZeroToHundred,
+  QuarterWidth,
+  TimeAxisMillisFromZero,
+  UsageAlarmFactory,
+  UsageCountThreshold,
+  UsageThreshold,
+} from "../../common";
+import {
+  MonitoringHeaderWidget,
+  MonitoringNamingStrategy,
+} from "../../dashboard";
+
+export interface AuroraClusterMonitoringOptions extends BaseMonitoringProps {
+  readonly addMaxServerlessDatabaseCapacityAlarm?: Record<
+    string,
+    UsageCountThreshold
+  >;
+  readonly addCpuUsageAlarm?: Record<string, UsageThreshold>;
+  readonly addMinConnectionCountAlarm?: Record<
+    string,
+    LowConnectionCountThreshold
+  >;
+  readonly addMaxConnectionCountAlarm?: Record<
+    string,
+    HighConnectionCountThreshold
+  >;
+}
+
+export interface AuroraClusterMonitoringProps
+  extends RdsClusterMetricFactoryProps,
+    AuroraClusterMonitoringOptions {}
+
+export class AuroraClusterMonitoring extends Monitoring {
+  readonly title: string;
+  readonly url?: string;
+
+  readonly usageAlarmFactory: UsageAlarmFactory;
+  readonly connectionAlarmFactory: ConnectionAlarmFactory;
+
+  readonly usageAnnotations: HorizontalAnnotation[];
+  readonly connectionAnnotations: HorizontalAnnotation[];
+
+  readonly connectionsMetric: MetricWithAlarmSupport;
+  readonly serverlessDatabaseCapacityMetric: MetricWithAlarmSupport;
+  readonly cpuUsageMetric: MetricWithAlarmSupport;
+  readonly selectLatencyMetric: MetricWithAlarmSupport;
+  readonly insertLatencyMetric: MetricWithAlarmSupport;
+  readonly updateLatencyMetric: MetricWithAlarmSupport;
+  readonly deleteLatencyMetric: MetricWithAlarmSupport;
+  readonly commitLatencyMetric: MetricWithAlarmSupport;
+
+  constructor(scope: MonitoringScope, props: AuroraClusterMonitoringProps) {
+    super(scope, props);
+
+    const metricFactory = new RdsClusterMetricFactory(
+      scope.createMetricFactory(),
+      props
+    );
+    this.connectionsMetric = metricFactory.metricTotalConnectionCount();
+    this.serverlessDatabaseCapacityMetric =
+      metricFactory.metricServerlessDatabaseCapacity();
```

Comment:
This would throw an error for non-`ServerlessCluster`s, so it should be guarded here in this class too.

Also missing tests that would've caught that.

Reply:
Ill add a test for the guard